lib/faktory_worker/worker.ex

```diff
@@ -51,6 +70,45 @@ defmodule FaktoryWorker.Worker do
 
   def send_beat(state), do: clear_beat_ref(state)
 
+  @spec send_fetch(state :: __MODULE__.t()) :: state :: __MODULE__.t()
+  def send_fetch(%{worker_state: worker_state} = state) when worker_state == :ok do
+    queues =
+      state.worker_config
+      |> Keyword.get(:queue, [])
+      |> format_queue_for_command()
+
+    state.conn
+    |> ConnectionManager.send_command({:fetch, queues})
+    |> handle_fetch_response(state)
+    |> schedule_fetch()
+  end
+
+  def send_fetch(state), do: state
+
+  @spec ack_job(state :: __MODULE__.t(), :ok | {:error, any()}) :: __MODULE__.t()
+  def ack_job(state, :ok) do
+    state.conn
+    |> ConnectionManager.send_command({:ack, state.job_id})
+    |> handle_ack_response(state)
+  end
+
+  def ack_job(state, {:error, reason}) do
+    backtrace_length = Keyword.get(state.worker_config, :backtrace, 30)
+
+    error = ErrorFormatter.format_error(reason, backtrace_length)
+
+    payload = %{
+      jid: state.job_id,
+      errtype: error.type,
+      message: error.message || "Undetected Error Type",
```

Comment:
When would there not be a message?

Reply:
Good spot, this should be on error type and moved to the error formatter.